base/strings/basic.jl

```diff
@@ -62,6 +62,7 @@ convert(::Type{Array{UInt8}}, s::AbstractString) = String(s).data
 convert(::Type{String}, s::AbstractString) = String(s)
 convert(::Type{Vector{Char}}, s::AbstractString) = collect(s)
 convert(::Type{Symbol}, s::AbstractString) = Symbol(s)
+convert(::Type{String}, s::Symbol) = unsafe_wrap(String, Cstring(s))
```

Comment:
Sorry, yes, this would means that all the strings have data that points to the same memory.

My suggestion is to use `unsafe_string` here, not, `unsafe_wrap`.   I don't think the extra efficiency of not making a copy is worth it for the danger of being able to modify `Symbol` contents through the `data` field of a string.